galaxy.py

```diff
@@ -23,23 +25,26 @@ def _get_ip():
     cmd_awk = ['awk', '{ print $2 }']
     p3 = subprocess.Popen(cmd_awk, stdin=p2.stdout, stdout=subprocess.PIPE)
     galaxy_ip = p3.stdout.read()
-    log.debug('Host IP determined to be %s', galaxy_ip)
     return galaxy_ip
 
 
-def _test_url(url, key, history_id):
+def _test_url(url, key, history_id, obj=True):
     """Test the functionality of a given galaxy URL, to ensure we can connect
     on that address."""
+    print url
     try:
-        gi = objects.GalaxyInstance(url, key)
-        gi.histories.get(history_id)
-        log.debug('Galaxy URL %s is functional', url)
+        if obj:
+            gi = objects.GalaxyInstance(url, key)
+            gi.histories.get(history_id)
+        else:
+            gi = GalaxyInstance(url=url, key=key)
+            gi.histories.get_histories(history_id)
         return gi
     except Exception:
         return None
 
 
-def get_galaxy_connection(history_id=None):
+def get_galaxy_connection(history_id=None, obj=True):
```

Comment:
Not sure, John and you pushed objects for a good reason. We should indicate that we want to use it here and everything else is a exception. However, there are only two functions and one is using them the other not.